[CI] RepositoryAzureClientYamlTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-periodic #3682 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/2is5kekrnhydy)
- [elasticsearch-pull-request #27743 / part-1-fips](https://gradle-enterprise.elastic.co/s/qzl4f6iowhjga)
- [elasticsearch-pull-request #27738 / part-1-fips](https://gradle-enterprise.elastic.co/s/5ym2kdu3xpoxm)
- [elasticsearch-pull-request #27717 / part-1-fips](https://gradle-enterprise.elastic.co/s/fmnilt6qv7qam)
- [elasticsearch-periodic #3678 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/mp7mbnd4hexjc)

**Reproduction Line:**
```
./gradlew ":modules:repository-azure:managedIdentityYamlRestTest" --tests "org.elasticsearch.repositories.azure.RepositoryAzureClientYamlTestSuiteIT.test {yaml=repository_azure/20_repository/teardown}" -Dtests.seed=E6F121B481805CC8 -Dtests.locale=mt -Dtests.timezone=Europe/Busingen -Druntime.java=17 -Dtests.fips.enabled=true
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.azure.RepositoryAzureClientYamlTestSuiteIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.azure.RepositoryAzureClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [repository_azure/20_repository:6]: expected [2xx] status code but api [snapshot.create_repository] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] path [_integration_tests] is not accessible on master node","stack_trace":"org.elasticsearch.repositories.RepositoryVerificationException: [repository] path [_integration_tests] is not accessible on master node\nCaused by: java.lang.IllegalArgumentException: Authority host must use https scheme.\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.util.ValidationUtil.validateAuthHost(ValidationUtil.java:87)\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.IdentityClientOptions.loadFromConfiguration(IdentityClientOptions.java:703)\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.IdentityClientOptions.<init>(IdentityClientOptions.java:89)\n\tat com.azure.identity@1.13.1/com.azure.identity.CredentialBuilderBase.<init>(CredentialBuilderBase.java:47)\n\tat com.azure.identity@1.13.1/com.azure.identity.DefaultAzureCredentialBuilder.<init>(DefaultAzureCredentialBuilder.java:77)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureClientProvider.createClient(AzureClientProvider.java:184)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureStorageService.client(AzureStorageService.java:91)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.getAzureBlobServiceClientClient(AzureBlobStore.java:633)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.asyncClient(AzureBlobStore.java:629)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.lambda$executeSingleUpload$19(AzureBlobStore.java:463)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.SocketAccess.lambda$doPrivilegedVoidException$0(SocketAccess.java:46)\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:569)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.SocketAccess.doPrivilegedVoidException(SocketAccess.java:45)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.executeSingleUpload(AzureBlobStore.java:462)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.writeBlob(AzureBlobStore.java:387)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlob(AzureBlobContainer.java:115)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlobAtomic(AzureBlobContainer.java:110)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:2156)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.repositories.RepositoriesService.lambda$validatePutRepositoryRequest$11(RepositoriesService.java:361)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:36)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"repository_verification_exception","reason":"[repository] path [_integration_tests] is not accessible on master node","caused_by":{"type":"illegal_argument_exception","reason":"Authority host must use https scheme.","stack_trace":"java.lang.IllegalArgumentException: Authority host must use https scheme.\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.util.ValidationUtil.validateAuthHost(ValidationUtil.java:87)\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.IdentityClientOptions.loadFromConfiguration(IdentityClientOptions.java:703)\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.IdentityClientOptions.<init>(IdentityClientOptions.java:89)\n\tat com.azure.identity@1.13.1/com.azure.identity.CredentialBuilderBase.<init>(CredentialBuilderBase.java:47)\n\tat com.azure.identity@1.13.1/com.azure.identity.DefaultAzureCredentialBuilder.<init>(DefaultAzureCredentialBuilder.java:77)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureClientProvider.createClient(AzureClientProvider.java:184)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureStorageService.client(AzureStorageService.java:91)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.getAzureBlobServiceClientClient(AzureBlobStore.java:633)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.asyncClient(AzureBlobStore.java:629)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.lambda$executeSingleUpload$19(AzureBlobStore.java:463)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.SocketAccess.lambda$doPrivilegedVoidException$0(SocketAccess.java:46)\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:569)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.SocketAccess.doPrivilegedVoidException(SocketAccess.java:45)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.executeSingleUpload(AzureBlobStore.java:462)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.writeBlob(AzureBlobStore.java:387)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlob(AzureBlobContainer.java:115)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlobAtomic(AzureBlobContainer.java:110)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:2156)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.repositories.RepositoriesService.lambda$validatePutRepositoryRequest$11(RepositoriesService.java:361)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:36)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"stack_trace":"org.elasticsearch.repositories.RepositoryVerificationException: [repository] path [_integration_tests] is not accessible on master node\nCaused by: java.lang.IllegalArgumentException: Authority host must use https scheme.\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.util.ValidationUtil.validateAuthHost(ValidationUtil.java:87)\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.IdentityClientOptions.loadFromConfiguration(IdentityClientOptions.java:703)\n\tat com.azure.identity@1.13.1/com.azure.identity.implementation.IdentityClientOptions.<init>(IdentityClientOptions.java:89)\n\tat com.azure.identity@1.13.1/com.azure.identity.CredentialBuilderBase.<init>(CredentialBuilderBase.java:47)\n\tat com.azure.identity@1.13.1/com.azure.identity.DefaultAzureCredentialBuilder.<init>(DefaultAzureCredentialBuilder.java:77)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureClientProvider.createClient(AzureClientProvider.java:184)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureStorageService.client(AzureStorageService.java:91)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.getAzureBlobServiceClientClient(AzureBlobStore.java:633)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.asyncClient(AzureBlobStore.java:629)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.lambda$executeSingleUpload$19(AzureBlobStore.java:463)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.SocketAccess.lambda$doPrivilegedVoidException$0(SocketAccess.java:46)\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:569)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.SocketAccess.doPrivilegedVoidException(SocketAccess.java:45)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.executeSingleUpload(AzureBlobStore.java:462)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobStore.writeBlob(AzureBlobStore.java:387)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlob(AzureBlobContainer.java:115)\n\tat org.elasticsearch.repository.azure@8.16.0-SNAPSHOT/org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlobAtomic(AzureBlobContainer.java:110)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:2156)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.repositories.RepositoriesService.lambda$validatePutRepositoryRequest$11(RepositoriesService.java:361)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:36)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":500}]
```

**Issue Reasons:**
- [main] 2 consecutive failures in class org.elasticsearch.repositories.azure.RepositoryAzureClientYamlTestSuiteIT
- [main] 2 consecutive failures in step openjdk17_checkpart1_java-fips-matrix
- [main] 3 consecutive failures in step part-1-fips
- [main] 5 failures in class org.elasticsearch.repositories.azure.RepositoryAzureClientYamlTestSuiteIT (2.6% fail rate in 194 executions)
- [main] 2 failures in step openjdk17_checkpart1_java-fips-matrix (100.0% fail rate in 2 executions)
- [main] 3 failures in step part-1-fips (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (9.7% fail rate in 31 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/repository-azure/src/yamlRestTest/java/org/elasticsearch/repositories/azure/RepositoryAzureClientYamlTestSuiteIT.java
+++ b/modules/repository-azure/src/yamlRestTest/java/org/elasticsearch/repositories/azure/RepositoryAzureClientYamlTestSuiteIT.java
@@ -14,6 +14,7 @@ import com.carrotsearch.randomizedtesting.annotations.Name;
 import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
 
 import org.elasticsearch.core.Booleans;
+import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.TestTrustStore;
 import org.elasticsearch.test.cluster.ElasticsearchCluster;
 import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
@@ -30,7 +31,9 @@ public class RepositoryAzureClientYamlTestSuiteIT extends ESClientYamlSuiteTestC
     private static final String AZURE_TEST_SASTOKEN = System.getProperty("test.azure.sas_token");
 
     private static AzureHttpFixture fixture = new AzureHttpFixture(
-        USE_FIXTURE ? AzureHttpFixture.Protocol.HTTPS : AzureHttpFixture.Protocol.NONE,
+        USE_FIXTURE
+            ? ESTestCase.inFipsJvm() ? AzureHttpFixture.Protocol.HTTP : AzureHttpFixture.Protocol.HTTPS
+            : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
         AZURE_TEST_CONTAINER,
         AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT)
@@ -59,7 +62,11 @@ public class RepositoryAzureClientYamlTestSuiteIT extends ESClientYamlSuiteTestC
             s -> USE_FIXTURE
         )
         .setting("thread_pool.repository_azure.max", () -> String.valueOf(randomIntBetween(1, 10)), s -> USE_FIXTURE)
-        .systemProperty("javax.net.ssl.trustStore", () -> trustStore.getTrustStorePath().toString(), s -> USE_FIXTURE)
+        .systemProperty(
+            "javax.net.ssl.trustStore",
+            () -> trustStore.getTrustStorePath().toString(),
+            s -> USE_FIXTURE && ESTestCase.inFipsJvm() == false
+        )
         .build();
 
     @ClassRule(order = 1)

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -113,18 +113,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/111308
 - class: org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT
   issue: https://github.com/elastic/elasticsearch/issues/111319
-- class: org.elasticsearch.xpack.searchablesnapshots.AzureSearchableSnapshotsIT
-  issue: https://github.com/elastic/elasticsearch/issues/111279
-- class: org.elasticsearch.repositories.azure.RepositoryAzureClientYamlTestSuiteIT
-  issue: https://github.com/elastic/elasticsearch/issues/111345
-- class: org.elasticsearch.xpack.repositories.metering.azure.AzureRepositoriesMeteringIT
-  issue: https://github.com/elastic/elasticsearch/issues/111307
-- class: org.elasticsearch.xpack.snapshotbasedrecoveries.recovery.AzureSnapshotBasedRecoveryIT
-  method: testRecoveryUsingSnapshots
-  issue: https://github.com/elastic/elasticsearch/issues/111377
-- class: org.elasticsearch.repositories.blobstore.testkit.AzureSnapshotRepoTestKitIT
-  method: testRepositoryAnalysis
-  issue: https://github.com/elastic/elasticsearch/issues/111280
 - class: org.elasticsearch.xpack.esql.analysis.VerifierTests
   method: testMatchFilter
   issue: https://github.com/elastic/elasticsearch/issues/111380

--- a/test/framework/src/main/java/org/elasticsearch/test/TestTrustStore.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/TestTrustStore.java
@@ -22,6 +22,7 @@ import java.util.List;
 import java.util.Objects;
 
 import static org.apache.lucene.tests.util.LuceneTestCase.createTempDir;
+import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 public class TestTrustStore extends ExternalResource {
@@ -35,6 +36,7 @@ public class TestTrustStore extends ExternalResource {
     private Path trustStorePath;
 
     public Path getTrustStorePath() {
+        assertFalse("Tests in FIPS mode cannot supply a custom trust store", ESTestCase.inFipsJvm());
         return Objects.requireNonNullElseGet(trustStorePath, () -> ESTestCase.fail(null, "trust store not created"));
     }
 

--- a/x-pack/plugin/repositories-metering-api/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/repositories/metering/azure/AzureRepositoriesMeteringIT.java
+++ b/x-pack/plugin/repositories-metering-api/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/repositories/metering/azure/AzureRepositoriesMeteringIT.java
@@ -10,6 +10,7 @@ import fixture.azure.AzureHttpFixture;
 
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Booleans;
+import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.TestTrustStore;
 import org.elasticsearch.test.cluster.ElasticsearchCluster;
 import org.elasticsearch.xpack.repositories.metering.AbstractRepositoriesMeteringAPIRestTestCase;
@@ -28,7 +29,9 @@ public class AzureRepositoriesMeteringIT extends AbstractRepositoriesMeteringAPI
     private static final String AZURE_TEST_SASTOKEN = System.getProperty("test.azure.sas_token");
 
     private static AzureHttpFixture fixture = new AzureHttpFixture(
-        USE_FIXTURE ? AzureHttpFixture.Protocol.HTTPS : AzureHttpFixture.Protocol.NONE,
+        USE_FIXTURE
+            ? ESTestCase.inFipsJvm() ? AzureHttpFixture.Protocol.HTTP : AzureHttpFixture.Protocol.HTTPS
+            : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
         AZURE_TEST_CONTAINER,
         AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT)
@@ -57,7 +60,11 @@ public class AzureRepositoriesMeteringIT extends AbstractRepositoriesMeteringAPI
             () -> "ignored;DefaultEndpointsProtocol=https;BlobEndpoint=" + fixture.getAddress(),
             s -> USE_FIXTURE
         )
-        .systemProperty("javax.net.ssl.trustStore", () -> trustStore.getTrustStorePath().toString(), s -> USE_FIXTURE)
+        .systemProperty(
+            "javax.net.ssl.trustStore",
+            () -> trustStore.getTrustStorePath().toString(),
+            s -> USE_FIXTURE && ESTestCase.inFipsJvm() == false
+        )
         .build();
 
     @ClassRule(order = 1)

--- a/x-pack/plugin/searchable-snapshots/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/searchablesnapshots/AzureSearchableSnapshotsIT.java
+++ b/x-pack/plugin/searchable-snapshots/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/searchablesnapshots/AzureSearchableSnapshotsIT.java
@@ -11,6 +11,7 @@ import fixture.azure.AzureHttpFixture;
 
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Booleans;
+import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.TestTrustStore;
 import org.elasticsearch.test.cluster.ElasticsearchCluster;
 import org.elasticsearch.test.rest.ESRestTestCase;
@@ -29,7 +30,9 @@ public class AzureSearchableSnapshotsIT extends AbstractSearchableSnapshotsRestT
     private static final String AZURE_TEST_SASTOKEN = System.getProperty("test.azure.sas_token");
 
     private static AzureHttpFixture fixture = new AzureHttpFixture(
-        USE_FIXTURE ? AzureHttpFixture.Protocol.HTTPS : AzureHttpFixture.Protocol.NONE,
+        USE_FIXTURE
+            ? ESTestCase.inFipsJvm() ? AzureHttpFixture.Protocol.HTTP : AzureHttpFixture.Protocol.HTTPS
+            : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
         AZURE_TEST_CONTAINER,
         AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT)

--- a/x-pack/plugin/snapshot-based-recoveries/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/snapshotbasedrecoveries/recovery/AzureSnapshotBasedRecoveryIT.java
+++ b/x-pack/plugin/snapshot-based-recoveries/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/snapshotbasedrecoveries/recovery/AzureSnapshotBasedRecoveryIT.java
@@ -11,6 +11,7 @@ import fixture.azure.AzureHttpFixture;
 
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Booleans;
+import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.TestTrustStore;
 import org.elasticsearch.test.cluster.ElasticsearchCluster;
 import org.junit.ClassRule;
@@ -28,7 +29,9 @@ public class AzureSnapshotBasedRecoveryIT extends AbstractSnapshotBasedRecoveryR
     private static final String AZURE_TEST_SASTOKEN = System.getProperty("test.azure.sas_token");
 
     private static AzureHttpFixture fixture = new AzureHttpFixture(
-        USE_FIXTURE ? AzureHttpFixture.Protocol.HTTPS : AzureHttpFixture.Protocol.NONE,
+        USE_FIXTURE
+            ? ESTestCase.inFipsJvm() ? AzureHttpFixture.Protocol.HTTP : AzureHttpFixture.Protocol.HTTPS
+            : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
         AZURE_TEST_CONTAINER,
         AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT)
@@ -59,7 +62,11 @@ public class AzureSnapshotBasedRecoveryIT extends AbstractSnapshotBasedRecoveryR
             s -> USE_FIXTURE
         )
         .setting("xpack.license.self_generated.type", "trial")
-        .systemProperty("javax.net.ssl.trustStore", () -> trustStore.getTrustStorePath().toString(), s -> USE_FIXTURE)
+        .systemProperty(
+            "javax.net.ssl.trustStore",
+            () -> trustStore.getTrustStorePath().toString(),
+            s -> USE_FIXTURE && ESTestCase.inFipsJvm() == false
+        )
         .build();
 
     @ClassRule(order = 1)

--- a/x-pack/plugin/snapshot-repo-test-kit/qa/azure/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/AzureSnapshotRepoTestKitIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/qa/azure/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/AzureSnapshotRepoTestKitIT.java
@@ -10,6 +10,7 @@ import fixture.azure.AzureHttpFixture;
 
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Booleans;
+import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.TestTrustStore;
 import org.elasticsearch.test.cluster.ElasticsearchCluster;
 import org.junit.ClassRule;
@@ -27,7 +28,9 @@ public class AzureSnapshotRepoTestKitIT extends AbstractSnapshotRepoTestKitRestT
     private static final String AZURE_TEST_SASTOKEN = System.getProperty("test.azure.sas_token");
 
     private static AzureHttpFixture fixture = new AzureHttpFixture(
-        USE_FIXTURE ? AzureHttpFixture.Protocol.HTTPS : AzureHttpFixture.Protocol.NONE,
+        USE_FIXTURE
+            ? ESTestCase.inFipsJvm() ? AzureHttpFixture.Protocol.HTTP : AzureHttpFixture.Protocol.HTTPS
+            : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
         AZURE_TEST_CONTAINER,
         AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT)
@@ -62,7 +65,11 @@ public class AzureSnapshotRepoTestKitIT extends AbstractSnapshotRepoTestKitRestT
                 c.systemProperty("test.repository_test_kit.skip_cas", "true");
             }
         })
-        .systemProperty("javax.net.ssl.trustStore", () -> trustStore.getTrustStorePath().toString(), s -> USE_FIXTURE)
+        .systemProperty(
+            "javax.net.ssl.trustStore",
+            () -> trustStore.getTrustStorePath().toString(),
+            s -> USE_FIXTURE && ESTestCase.inFipsJvm() == false
+        )
         .build();
 
     @ClassRule(order = 1)

